src/luarocks/persist.lua

```diff
@@ -147,17 +144,11 @@ write_table = function(out, tbl, level, field_order)
       end
 
       write_value(out, v, level, sub_order)
-      if type(k) == "number" then
```

Comment:
I was looking at the manifest tables. I suggest this compromise solution: `if type(v) == "number"`. This will catch the version tables correctly.